src/bun.js/bindings/napi.cpp

```diff
@@ -1502,65 +1677,17 @@ JSC_DEFINE_HOST_FUNCTION(NapiClass_ConstructorFunction,
     callFrame->setThisValue(subclass);
 
     size_t count = callFrame->argumentCount();
-    MarkedArgumentBuffer args;
-
-    if (count > 6) {
-        for (size_t i = 6; i < count; i++) {
-            args.append(callFrame->uncheckedArgument(i));
-        }
-    }
+    MarkedArgumentBufferWithSize<12> args;
+    size_t argc = callFrame->argumentCount() + 1;
+    args.fill(vm, argc, [&](auto* slot) {
+        memcpy(slot, callFrame->addressOfArgumentsStart() - 1, sizeof(JSC::JSValue) * argc);
```

Comment:
we do the -1 offset to copy this into slot?

Reply:
yeah i should write a comment here for this

its the `this` value's slot